Firestore/Example/Tests/Integration/API/FIRFieldsTests.mm

```diff
@@ -221,28 +225,83 @@ - (void)testFieldsWithSpecialCharsCanBeUsedInOrderBy {
   [self awaitExpectations];
 }
 
-- (NSDictionary<NSString *, id> *)testDataWithTimestamp:(FIRTimestamp *)timestamp {
-  return @{
-    @"timestamp" : [timestamp approximateDateValue],
-    @"metadata" : @{@"nestedTimestamp" : [timestamp approximateDateValue]}
-  };
+- (FIRDocumentSnapshot *)snapshotWithTimestamps:(FIRTimestamp *)timestamp {
+  FIRDocumentReference *doc = [self documentRef];
+  NSDictionary<NSString *, id> *data =
+      @{ @"timestamp" : timestamp,
+         @"nested" : @{@"timestamp2" : timestamp} };
+  [self writeDocumentRef:doc data:data];
+  return [self readDocumentForRef:doc];
 }
 
-// This test should break once the default for how timestamps are returned changes.
-- (void)testThatDataContainsNativeDateType {
-  NSDate *date = [NSDate date];
-  FIRTimestamp *timestamp = [FIRTimestamp timestampWithDate:date];
+// Note: timestampsInSnapshotsEnabled is set to "true" in FSTIntegrationTestCase, so this test is
+// not affected by the current default in FIRFirestoreSettings.
+- (void)testTimestampsInSnapshots {
+  FIRTimestamp *originalTimestamp = [FIRTimestamp timestampWithSeconds:100 nanoseconds:123456789];
   FIRDocumentReference *doc = [self documentRef];
-  [self writeDocumentRef:doc data:[self testDataWithTimestamp:timestamp]];
+  [self writeDocumentRef:doc data:testDataWithTimestamps(originalTimestamp)];
+
+  FIRDocumentSnapshot *snapshot = [self readDocumentForRef:doc];
+  NSDictionary<NSString *, id> *data = [snapshot data];
+  // Timestamp are currently truncated to microseconds after being written to the database.
+  FIRTimestamp *truncatedTimestamp =
+      [FIRTimestamp timestampWithSeconds:originalTimestamp.seconds
+                             nanoseconds:originalTimestamp.nanoseconds / 1000 * 1000];
+
+  FIRTimestamp *timestampFromSnapshot = snapshot[@"timestamp"];
+  FIRTimestamp *timestampFromData = data[@"timestamp"];
+  XCTAssertEqualObjects(truncatedTimestamp, timestampFromData);
+  XCTAssertEqualObjects(timestampFromSnapshot, timestampFromData);
+
+  timestampFromSnapshot = snapshot[@"nested.timestamp2"];
+  timestampFromData = data[@"nested"][@"timestamp2"];
+  XCTAssertEqualObjects(truncatedTimestamp, timestampFromData);
+  XCTAssertEqualObjects(timestampFromSnapshot, timestampFromData);
+}
+@end
 
-  FIRDocumentSnapshot *result = [self readDocumentForRef:doc];
-  NSDate *resultDate = result.data[@"timestamp"];
-  XCTAssertEqualWithAccuracy([resultDate timeIntervalSince1970], [date timeIntervalSince1970],
-                             0.000001);
-  XCTAssertEqualObjects(result.data[@"timestamp"], resultDate);
-  NSDate *resultNestedDate = result[@"metadata.nestedTimestamp"];
-  XCTAssertEqualWithAccuracy([resultNestedDate timeIntervalSince1970], [date timeIntervalSince1970],
-                             0.000001);
+// Settings can only be redefined before client is initialized, so this has to happen in setUp.
```

Comment:
This comment seems kind of out of place. If we have it at all, it should perhaps be in the setUp method? :-)

Reply:
Done.